actions/node_discover_and_add_interfaces_by_name_and_type.py

```diff
@@ -0,0 +1,111 @@
+           interface_names: List of ifName values to filter add interfaces
+           interface_type: List of ifType values to filter add interfaces
+
+        """
+        results = {'added': [], 'existing': [], 'removed': []}
+
+        self.connect()
+
+        orion_node = self.get_node(node)
+
+        if not orion_node.npm:
+            raise ValueError("Node not found")
+
+        Discoverdinterfaces = self.invoke('Orion.NPM.Interfaces',
+                                          'DiscoverInterfacesOnNode',
+                                          orion_node.npm_id)
+
+        add_interfaces = []
+        for interface in Discoverdinterfaces['DiscoveredInterfaces']:
+            # Unmonitored interfaces have an InterfaceID of 0.
+            if not interface['InterfaceID'] == 0:
+                self.logger.info("Skipping {} as monitored (I:{})".format(
+                    interface['Caption'],
+                    interface['InterfaceID']))
+                results['existing'].append(
+                    {interface['Caption']: interface['InterfaceID']})
+            else:
+                self.logger.info("Interface {} NOT monitored. (ID:{} Type:{} \
+                Admin Status:{} )".format(
+                    interface['Caption'],
+                    interface['InterfaceID'],
+                    interface['ifType'],
+                    interface['ifAdminStatus']))
+
+                # admin_up_required is defined as boolean in YAML file
+                if admin_up_required:
+                    # Loop through values in interface_type list, compare to Discovered interface
+                    # data and add interfaces with IfType values that match what is provided in list
+                    # and have the IfAdminStatus value as "UP" or 1
+                    for iftype in interface_type:
+                        if interface['ifAdminStatus'] == 1 and interface['ifType'] == iftype:
+                            self.logger.info('Interface: {} is of IfType: {}. Adding...'.format(
+                                interface['Caption'],
+                                interface['ifType']))
+                            add_interfaces.append(interface)
+                else:
+                    # Loop through values in interface_type list, compare to Discovered interface
+                    # data and add interfaces with IfType values that match what is provided in list
+                    # without checking the IfAdminStatus value
+                    for iftype in interface_type:
+                        if interface['ifType'] == iftype:
+                            self.logger.info('Interface: {} is of IfType: {}. Adding...'.format(
+                                interface['Caption'],
+                                interface['ifType']))
+                            add_interfaces.append(interface)
+        self.logger.info('Adding interfaces to monitoring config...')
```

Comment:
looks like the formatting is off here

Reply:
PyCharm is not complaining about the formatting here.  What exactly is wrong?  PEP8 requires line 78 to be less than 100 characters so 79 and 80 are moved down there to address that.